### PR TITLE
Adding some sort of CN check

### PR DESCRIPTION
**Pull request by matzew**

### CODE REVIEW

**Comment by matzew** [comment]
@pb82 mind check here ?

**Comment by pb82** [comment]
👀

File: push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java [hunk 1]
```diff
@@ -79,12 +86,22 @@ public static boolean checkValidity(final byte[] keystore, final char[] password
...
                 final X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
+                final X500Principal subjectX500Principal = certificate.getSubjectX500Principal();
+                final String subject = subjectX500Principal.getName();
 
-                try{
-                    certificate.checkValidity();
-                } catch (CertificateExpiredException | CertificateNotYetValidException e) {
-                    LOGGER.error("Provided APNs .p12 file is expired or not yet valid");
-                    return false;
+                final Matcher matcher = COMMON_NAME_PATTERN.matcher(subject);
```
Comment (ziccardi) [inline push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java:92]: I believe we have bouncycastle in the list of UPS dependencies. If that is the case, a more robust way to extract the CN would be:
```java
X500Name x500name = new JcaX509CertificateHolder(cert).getSubject();
RDN cn = x500name.getRDNs(BCStyle.CN)[0];
```

Comment (pb82) [reply push-sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java:92]: @ziccardi thanks, i will try this

**Comment by pb82** [comment]
@ziccardi @matzew sent a PR to this PR: https://github.com/matzew/aerogear-unifiedpush-server/pull/18

Does that make sense to you?

**Review by ziccardi — APPROVED** [review]
LGTM

**Review by pb82 — APPROVED** [review]

**Comment by pb82** [comment]
@matzew can you please merge this?

**Pull request merged**